docs/reference/analysis/tokenfilters/unique-tokenfilter.asciidoc

```diff
@@ -4,7 +4,147 @@
 <titleabbrev>Unique</titleabbrev>
 ++++
 
-The `unique` token filter can be used to only index unique tokens during
-analysis. By default it is applied on all the token stream. If
-`only_on_same_position` is set to `true`, it will only remove duplicate
-tokens on the same position.
+Removes duplicate tokens from a stream. For example, you can use the `unique`
+filter to change `the lazy lazy dog` to `the lazy dog`.
+
+If the `only_on_same_position` parameter is set to `true`, the `unique` filter
+removes only duplicate tokens _in the same position_.
+
+[NOTE]
+====
+When `only_on_same_position` is `true`, the `unique` filter works the same as
+<<analysis-remove-duplicates-tokenfilter,`remove_duplicates`>> filter.
+====
```

Comment:
@romseygeek Do you know if there's any reason for users to prefer one of these filters over another? If so, I can note that here.

Reply:
Sorry, missed this! `unique` has the ability to work over an entire token stream, while `remove_duplicates` looks only at consecutive tokens, but other than that they are identical.